Python/Product/TestAdapter.Executor/Pytest/TestExecutorPytest.cs

```diff
@@ -132,22 +129,46 @@ IFrameworkHandle frameworkHandle
             IFrameworkHandle frameworkHandle
         ) {
             PythonProjectSettings settings = testGroup.Key;
-            if (settings.TestFramwork != TestFrameworkType.Pytest) {
+            if (settings == null || settings.TestFramwork != TestFrameworkType.Pytest) {
                 return;
             }
 
             using (var executor = new ExecutorService(settings, frameworkHandle, runContext)) {
+                Dictionary<string, TestResult> idToResultsMap = CreatePytestIdToVsTestResultsMap(testGroup);
```

Comment:
Previous code in TestResultParser.Parse(..) stored a list of TestResults create from Testcases, but these lead to linear searching through